gamd/gamdSimulation.py

```diff
@@ -112,25 +112,61 @@ def createGamdSimulation(self, config, platform_name, device_index):
                 raise Exception("Invalid input type: %s. Allowed types are: "\
                                 "'pdb' and 'rst7'/'inpcrd'.")
             gamdSimulation.system = prmtop.createSystem(
-                nonbondedMethod=nonbondedMethod, 
-                nonbondedCutoff=config.system.nonbonded_cutoff, 
+                nonbondedMethod=nonbondedMethod,
+                nonbondedCutoff=config.system.nonbonded_cutoff,
                 constraints=constraints)
-            
+
+
         elif config.input_files.charmm is not None:
+
+            def read_toppar(filename):
+                extlist = ['rtf', 'prm', 'str']
+
+                parFiles = ()
+                for line in open(filename, 'r'):
+                    if '!' in line: line = line.split('!')[0]
+                    parfile = line.strip()
+                    if len(parfile) != 0:
+                        ext = parfile.lower().split('.')[-1]
+                        if not ext in extlist: continue
+                        parFiles += ( parfile, )
+
+                params = openmm_app.CharmmParameterSet( *parFiles )
+                return params
+
+            def gen_box(psf, boxsize):
+
+                boxvectors = boxsize
+
+                boxlx = boxvectors[0]*unit.nanometer
+                boxly = boxvectors[1]*unit.nanometer
+                boxlz = boxvectors[2]*unit.nanometer
+
+                psf.setBox(boxlx, boxly, boxlz)
+                return psf
+
+
+
             psf = openmm_app.CharmmPsfFile(
                 config.input_files.charmm.topology)
-            pdb_coords_filename = config.input_files.charmm.coordinates
-            positions, box_vectors = load_pdb_positions_and_box_vectors(
-                pdb_coords_filename, need_box)
-            params = openmm_app.CharmmParameterSet(
-                *config.input_files.charmm.parameters)
+            positions = openmm_app.CharmmCrdFile(config.input_files.charmm.coordinates)
+            box_vector = float(config.input_files.charmm.box_vector)
+            box_vectors = [box_vector, box_vector, box_vector]
+
+            psf = gen_box(psf, box_vectors)
+            toppar = config.input_files.charmm.parameters
+            charmm_params = read_toppar(toppar)
+
             topology = psf
             gamdSimulation.system = psf.createSystem(
-                params=params,
-                nonbondedMethod=nonbondedMethod, 
-                nonbondedCutoff=config.system.nonbonded_cutoff, 
+                params=charmm_params,
+                nonbondedMethod=nonbondedMethod,
+                nonbondedCutoff=config.system.nonbonded_cutoff,
+                switchDistance=1.0*unit.nanometer,
+                ewaldErrorTolerance = 0.0005,
```

Comment:
Values should be defaulted in the configuration and have XML tags made available to override the value.